### ISSUE TITLE
groupBy support

### ISSUE THREAD

**Issue by ThatJoeMoore** [morintd/prismock#670]
This isn't so much a feature request as much as it is an offer to take a crack at implementing it, as it would be super useful for my team.  Obviously groupBy is pretty complex, but are there any pitfalls besides that?

**Comment by morintd** [comment]
Hello @ThatJoeMoore , thanks a lot for your interest in Prismock! I'm definitely open to receiving PRs to add such features.

As you said, groupBy is pretty complex, I gave it a try at the beginning of Prismock but chose to pursue other features first as there was a lot of use-cases. I don't see any particular pitfalls if you want to implement groupBy outside of its own complexity.

Maybe, the most difficult part would be understanding the current code and writing the necessary tests. Feel free to let me know if you have any question, would love to discuss

**Comment by ThatJoeMoore** [comment]
I spent some time on this last night, and two things to say:

1) actually loving how the project is set up and structured. Took a sec to get into it, but I like it.
2) implementing other stuff first was 100% the right call. I'm finding that, now that those things are there, most of groupBy is just composing the other functions in new and interesting ways.

I'll probably send you a draft PR tonight with the initial bits - I've got tests passing on postgres for a simple `where ... group by` query and I can see how to make `having` and `order by` work by reusing the existing functions.
